app/controllers/application_controller.rb

```diff
@@ -13,11 +15,48 @@ def set_locale
              else
                http_accept_language.compatible_language_from(I18n.available_locales)
              end
+
     if locale && I18n.available_locales.include?(locale.to_sym)
       session[:locale] = I18n.locale = locale.to_sym
     end
   end
 
+  def bootstrap_class_for(flash_type)
+    case flash_type.to_s
+    when "success"
+      "alert-success"
+    when "error"
+      "alert-danger"
+    when "alert"
+      "alert-warning"
+    when "notice"
+      "alert-info"
+    else
+      flash_type.to_s
+    end
+  end
+
+  def flash_to_headers
+    return unless request.xhr?
+    response.headers['X-Message'] = flash_message
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.